frontend/www/js/omegaup/components/course/StudentProgress.vue

```diff
@@ -0,0 +1,103 @@
+<template>
+  <tr>
+    <td class="text-center align-middle">
+      <a v-bind:href="studentProgressUrl()">
+        {{ student.name || student.username }}
+      </a>
+    </td>
+    <td
+      class="score flex-column justify-content-center align-items-center"
+      v-for="assignment in assignments"
+    >
+      <p class="mb-1 text-center">{{ Math.round(score(assignment)) }}%</p>
+      <div class="d-flex justify-content-center">
+        <div v-if="student.progress.hasOwnProperty(assignment.alias) == false">
+          {{ T.wordsProblemsUnsolved }}
+        </div>
+        <div
+          v-for="problem in student.progress[assignment.alias]"
+          v-bind:class="getProblemColor(Math.round(problem))"
+          data-toggle="tooltip"
+          data-placement="bottom"
+          v-bind:title="problemScore(problem)"
+        ></div>
+      </div>
+    </td>
+  </tr>
+</template>
+
+<style>
+.box {
+  width: 20px;
+  height: 20px;
+  border: 1px solid black;
+  margin: -0.5px;
+}
+
+.bg-green {
+  background: rgb(53, 184, 53);
+}
+
+.bg-yellow {
+  background: yellow;
+}
+
+.bg-red {
+  background: red;
+}
+
+.bg-black {
+  background: rgb(53, 53, 53);
+}
+</style>
+
+<script lang="ts">
+import { Vue, Component, Prop, Watch } from 'vue-property-decorator';
+import { omegaup } from '../../omegaup';
+import { types } from '../../api_types';
+import T from '../../lang';
+
+@Component
+export default class StudentProgress extends Vue {
+  @Prop() course!: types.CourseDetails;
+  @Prop() student!: types.StudentProgress;
+  @Prop() assignments!: omegaup.Assignment[];
+
+  T = T;
+
+  problemScore(problem: number): string {
+    return Math.round(problem) + '%';
```

Comment:
nit: the use of [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) is preferred over concatenation with `+`:

```suggestion
    return `{Math.round(problem)}%`;
```

(although this was suggested to be inlined above, so this method can be removed).